### DISCUSSION TITLE
Examples of a context menu on a node?

### DISCUSSION THREAD

**Discussion by cgen01** [xyflow/xyflow#3227] [General]
Is there an example of using the `onNodeContextMenu` to position a component as a menu inside the react flow grid? I’m having the hardest time finding a working example with an actual context menu. I’m trying to implement a right click menu on a node inside the react flow grid. Thank you for any help with this!

**Comment by hayleigh-dot-dev** [comment]
Heya @cgen01 👋 What trouble are you running into exactly? The `onNodeContextMenu` handler is given the mouse event so you should be able to position your context menu using some of the properties on the event: `clientX/Y`, `offsetX/Y`, or `pageX/Y`.

Let me try and put together a quick example for you.

**Reply by hayleigh-dot-dev** [reply]
Here's a little codesandbox showing one approach. Hope this helps! https://codesandbox.io/s/node-context-menu-cnx3zt

**Comment by nandorojo** [comment]
Is there an example of context menu on the root flow canvas?

**Reply by meronogbai** [reply]
You can use `onPaneContextMenu` instead of `onNodeContextMenu` in the example above!
